modules/stereo/src/descriptor.cpp

```diff
@@ -0,0 +1,150 @@
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+//  * Neither the names of the copyright holders nor the names of the contributors
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+//
+//This software is provided by the copyright holders and contributors "as is" and
+//any express or implied warranties, including, but not limited to, the implied
+//warranties of merchantability and fitness for a particular purpose are disclaimed.
+//In no event shall copyright holders or contributors be liable for any direct,
+//indirect, incidental, special, exemplary, or consequential damages
+//(including, but not limited to, procurement of substitute goods or services;
+//loss of use, data, or profits; or business interruption) however caused
+//and on any theory of liability, whether in contract, strict liability,
+//or tort (including negligence or otherwise) arising in any way out of
+//the use of this software, even if advised of the possibility of such damage.
+
+/*****************************************************************************************************************\
+*   The file contains the implemented descriptors                                                                  *
+\******************************************************************************************************************/
+#include "descriptor.hpp"
+
+using namespace cv;
+using namespace stereo;
+
+//function that performs the census transform on two images.
+//Two variants of census are offered a sparse version whcih takes every second pixel as well as dense version
+void cv::stereo::censusTransform(const cv::Mat &image1, const cv::Mat &image2, int kernelSize, cv::Mat *&dist, const int type, const int numImages)
+{
+    CV_Assert(image1.size() == image2.size());
+    CV_Assert(kernelSize % 2 != 0);
+    CV_Assert(image1.type() == CV_8UC1 && image2.type() == CV_8UC1);
+    CV_Assert(type < 2 && type >= 0);
+    CV_Assert(kernelSize <= ((type == 0) ? 5 : 11));
+    int n2 = (kernelSize) / 2;
+    uint8_t *images[] = {image1.data, image2.data};
+    int *costs[] = {(int *)dist[0].data,(int *)dist[1].data};
+    if(type == CV_DENSE_CENSUS)
+    {
+        parallel_for_(cv::Range(n2, image1.rows - n2),
+            CombinedDescriptor<1,1,1,CV_DUAL_DESCRIPTOR,CensusKernel>(image1.cols, image1.rows,n2,costs,CensusKernel(images, numImages),n2));
+    }
+    else if(type == CV_SPARSE_CENSUS)
+    {
+        parallel_for_(cv::Range(n2, image1.rows - n2),
+            CombinedDescriptor<2,2,1,CV_DUAL_DESCRIPTOR,CensusKernel>(image1.cols, image1.rows,n2,costs,CensusKernel(images, numImages),n2));
+    }
+}
+//in a 9x9 kernel only certain positions are choosen for comparison
+void cv::stereo::starCensusTransform(const cv::Mat &img1, const cv::Mat &img2, int kernelSize, cv::Mat *&dist,const int numImages)
```

Comment:
There seem to be only 2-image functions. Do you plan to add single-image functions later?